app/views/sites/_form.html.erb

```diff
@@ -0,0 +1,70 @@
+  <% end %>
+  <%= f.text_field :location_name, class: "govuk-input govuk-!-width-full", aria: { describedby: @errors[:location_name] ? "location_name-error" : '' }, data: { qa: 'location__name' } %>
+</div>
+
+<h2 class="govuk-heading-m govuk-!-margin-top-8">Address</h2>
+
+<div class="govuk-form-group<%= " govuk-form-group--error" if @errors[:address1]&.any? %>">
+  <%= f.label :address1, class: "govuk-label" do %>
+    Building and street
+    <span class="govuk-visually-hidden">line 1 of 2</span>
+  <% end %>
+  <% if @errors[:address1]&.any? then %>
+    <span id="address1-error" class="govuk-error-message">
+      <span class="govuk-visually-hidden">Error:</span> <%= @errors[:address1]&.first %>
+    </span>
+  <% end %>
+  <%= f.text_field :address1, class: "govuk-input govuk-!-width-full", aria: { describedby: @errors[:address1] ? "address1-error" : '' } %>
+</div>
+
+<div class="govuk-form-group">
+  <%= f.label :address1, class: "govuk-label" do %>
+    <span class="govuk-visually-hidden">line 2 of 2</span>
+  <% end %>
+  <%= f.text_field :address2, class: "govuk-input govuk-!-width-full" %>
+</div>
+
+<div class="govuk-form-group<%= " govuk-form-group--error" if @errors[:address3]&.any? %>">
+  <%= f.label :address3, "Town or city", class: "govuk-label" %>
+  <% if @errors[:address3]&.any? then %>
+    <span id="address3-error" class="govuk-error-message">
+      <span class="govuk-visually-hidden">Error:</span> <%= @errors[:address3]&.first %>
+    </span>
+  <% end %>
+  <%= f.text_field :address3, class: "govuk-input govuk-!-width-full", aria: { describedby: @errors[:address3] ? "address3-error" : '' } %>
+</div>
+
+<div class="govuk-form-group">
+  <%= f.label :address4, "County", class: "govuk-label" %>
+  <%= f.text_field :address4, class: "govuk-input govuk-!-width-full" %>
+</div>
+
+<div class="govuk-form-group<%= " govuk-form-group--error" if @errors[:postcode]&.any? %>">
+  <%= f.label :postcode, class: "govuk-label" %>
+  <% if @errors[:postcode]&.any? then %>
+    <span id="postcode-error" class="govuk-error-message">
+      <span class="govuk-visually-hidden">Error:</span> <%= @errors[:postcode]&.first %>
+    </span>
+  <% end %>
+  <%= f.text_field :postcode, class: "govuk-input form-control-small", aria: { describedby: @errors[:postcode] ? "postcode-error" : '' } %>
+</div>
+
+<div class="govuk-form-group<%= " govuk-form-group--error" if @errors[:region_code]&.any? %>">
+  <%= f.label :region_code, "Region of UK", class: "govuk-label" %>
+  <% if @errors[:region_code]&.any? then %>
+    <span id="region_code-error" class="govuk-error-message">
+      <span class="govuk-visually-hidden">Error:</span> <%= @errors[:region_code]&.first %>
+    </span>
+  <% end %>
+  <%= f.select :region_code, Site::REGIONS, {}, class: "govuk-select", aria: { describedby: @errors[:region_code] ? "region_code-error": '' } %>
+</div>
+
+<%= f.submit "Publish changes", class: "govuk-button", data: { qa: 'location__publish-changes' } %>
```

Comment:
When adding a location the button text should be: "Save"
When editing a location the button text should be "Save and publish changes"

(Technically the new location will be published, but as it's not attached to any course it's not visible anywhere and so publish can be confusing)

Reply:
Done